apps/web/modules/bookings/views/bookings-listing-view.tsx

```diff
@@ -144,6 +145,23 @@ export default function Bookings() {
     )[0] || [];
 
   const [animationParentRef] = useAutoAnimate<HTMLDivElement>();
+  const { buttonToShow, isLoading, enableNotifications, disableNotifications } = useNotifications();
+
+  const actions = (
+    <div>
+      {buttonToShow && (
+        <Button
+          color="primary"
+          onClick={buttonToShow === "allow" ? enableNotifications : disableNotifications}
+          loading={isLoading}
+          disabled={buttonToShow === "denied"}
+          tooltipSide="bottom"
+          tooltip={buttonToShow === "denied" ? t("you_have_denied_notifications") : undefined}>
+          {t(buttonToShow === "disable" ? "disable_browser_notifications" : "allow_browser_notifications")}
+        </Button>
+      )}
+    </div>
+  );
```

Comment:
A JSX component shouldn't be defined within another component. It would cause it to loose its state on every render.

Reply:
It's being passed to Shell component so I guess it's fine doing it here. Plus that's how it is done in every other place as well.

Reply:
It does make sense; we have to follow best practices

Reply:
@anikdhabal I think it's fine because we need `buttonToShow`, `enableNotifications` which cannot be accessed outside of Bookings component and actions is not a big component that re rendering would cause any performance issue.